### DISCUSSION TITLE
Multi Tenant - Need some advice for my scenario

### DISCUSSION THREAD

**Discussion by KhaledGabr** [directus/directus#5014] [Archive]
I am exploring if Directus will be a good fit for this scenario.  I am building a platform where Course creators can manage their own **courses** and **students**.

**Project Scoping**
Project scoping seems to fit well. Since, each tenants data of each course creator is totally separate from others, and infrastructure should not be painful with tools like docker.

However, My concern is how hard will it be to update dashboard settings and schema changes to all my tenants as I am releasing new features.

I guess my question sort of boils down to Environment migration.
1) Are schema changes getting tracked somewhere and can be applied easily to other environments
2) How easy will it be to migrate **settings** ( and NOT content, users, or assets  ) to other environments

Thanks

**Comment by danieljwestman** [comment]
This is an _"obstacle"_ for us too. Our plan is to build a hook that in some valid cases creates corresponding custom migrations for schema/collection/field changes. These migrations can then be run in every environment (in our case automatic deploy on DO App Platform with build commands).

Also, I know there is new export/import options on the Directus core roadmap.

**Reply by benhaynes** [reply]
I believe import/export has been prioritized and will be started this week.
